src/components/user/uc/users.uc.js

```diff
@@ -0,0 +1,67 @@
+const { ObjectId } = require('mongoose').Types;
+const { GeneralError, NotFound } = require('../../../errors');
+const { userRepo, accountRepo, trashbinRepo } = require('../repo/index');
+const restrictToSameSchool = require('../../../utils/restrictToSameSchool');
+const restrictToRole = require('../../../utils/restrictToRole');
+
+const getUserData = async (id, app) => {
+	const data = {};
+
+	const user = await userRepo.getUser(id, app);
+	if (!(user && user._id && !user.deletedAt)) {
+		throw new NotFound(`User ${id} not found`);
+	}
+	data.user = user;
+
+	const account = await accountRepo.getUserAccount(id, app);
+	if (account) {
+		data.account = account;
+	}
+	return data;
+};
+
+const deleteUserData = async (id, app) => {
+	await accountRepo.deleteUserAccount(id, app);
+};
+
+const createUserTrashbin = async (id, data) => {
+	const trashBin = await trashbinRepo.createUserTrashbin(id, data);
+	if (!(trashBin && trashBin._id)) {
+		throw new GeneralError(`Unable to initiate trashBin`);
+	}
+	return trashBin;
+};
+
+const replaceUserWithTombstone = async (id, app) => {
+	const uid = ObjectId();
+	await userRepo.replaceUserWithTombstone(
+		id,
+		{
+			firstName: 'DELETED',
+			lastName: 'USER',
+			email: `${uid}@deleted`,
+			deletedAt: new Date(),
+		},
+		app
+	);
+	return { success: true };
+};
+
+const deleteUserUC = async (id, roleName, { account, app }) => {
+	await restrictToSameSchool(id, account, app);
+	await restrictToRole(id, roleName, app);
+
+	const data = await getUserData(id, app);
```

Comment:
Include to many keys in user object at the moment and rename from data to explicit what is saved will helpful. Later we add typescript and must define a schema what data is.